mlflow/utils/environment.py

```diff
@@ -18,11 +25,131 @@
+        self.dependencies = dependencies
+
+    def __str__(self):
+        return str(self.to_dict())
+
+    @classmethod
+    def default(cls):
+        return cls(
+            python=cls._get_current_python(),
+            build_dependencies=cls.get_default_build_dependencies(),
+            dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
+        )
+
+    @staticmethod
+    def _get_current_python():
+        return ".".join(map(str, sys.version_info[:3]))
+
+    @staticmethod
+    def _get_package_version(package_name):
+        try:
+            return __import__(package_name).__version__
+        except (ImportError, AttributeError):
+            return None
+
+    @staticmethod
+    def get_default_build_dependencies():
+        build_dependencies = []
+        for package in ["pip", "setuptools", "wheel"]:
+            version = PythonEnv._get_package_version(package)
+            dep = (package + "==" + version) if version else package
+            build_dependencies.append(dep)
+        return build_dependencies
+
+    def to_dict(self):
+        return self.__dict__.copy()
+
+    @classmethod
+    def from_dict(cls, dct):
+        return cls(**dct)
+
+    def to_yaml(self, path):
+        with open(path, "w") as f:
+            # Exclude None and empty lists
+            data = {k: v for k, v in self.to_dict().items() if v}
+            yaml.safe_dump(data, f, sort_keys=False, default_flow_style=False)
+
+    @classmethod
+    def from_yaml(cls, path):
+        with open(path) as f:
+            return cls.from_dict(yaml.safe_load(f))
+
+    @classmethod
+    def from_conda_yaml(cls, path):
+        with open(path) as f:
+            conda_env = yaml.safe_load(f)
+
+        python = None
+        pip_dependencies = None
+        build_dependencies = None
+        for dep in conda_env.get("dependencies", []):
+            if isinstance(dep, str):
+                if not python:
+                    match = _PYTHON_REGEX.match(dep)
+                    if match:
+                        python = match.group(2)
+                        continue
+
+                match = _BUILD_PACKAGE_REGEX.match(dep)
+                if match:
+                    if build_dependencies is None:
+                        build_dependencies = []
+                    name, operator, version = match.groups()
+                    operator = "==" if operator == "=" else operator
+                    build_dependencies.append(name + operator + version)
+            elif _is_pip_deps(dep):
+                pip_dependencies = dep["pip"]
+            else:
+                raise MlflowException(
+                    f"Invalid conda dependency: {dep}. Must be str or dict in the form of "
+                    '{"pip": [...]}',
+                )
+
+        if python is None:
+            raise MlflowException(
+                f"Failed to construct `PythonEnv`. {path} is missing a conda dependency for "
+                "python (e.g. 'python=3.8').",
```

Comment:
```suggestion
                "python (e.g. 'python=3.8').",
                error_code=INVALID_PARAMETER_VALUE
```

Is `INVALID_PARAMETER_VALUE` appropriate for this error?

Reply:
Maybe yes? because this is caused by an invalid parameter value (= conda.yaml content)